### ISSUE TITLE
Are counterblock + counterwallet still needed?

### ISSUE THREAD

**Issue by jotapea** [CounterpartyXCP/federatednode#341]
We should consider removing counterblock and counterwallet from the fednode software. Trim out the fat, these are not required to run a Counterparty node and add complexity and confusion to new developers.

**Comment by jdogresorg** [comment]
Counterwallet and counterblock are still the "official" wallets and are included in the fednode stack as an optional install. This allows anyone to spin up a FULL fednode which includes a locally run wallet service (counterwallet) and do their own transactions in a local wallet and does not require and external services / API calls.

Users can easily install just the base fednode stack (bitcoin, addrindexrs, counterparty-lib) by running:

`fednode install base master`

At some point in the future when there is a fully open source wallet alternative that works for desktop, mobile, and browsers, then we can consider depreciating counterwallet and counterblock and removing them from the fednode stack.

**Comment by jotapea** [comment]
Should a wallet be part of the Counterparty protocol stack?

I believe this is a valid question, as counterparty-lib can actually create ANY transaction without involving the private keys of the user. The signing of this transaction is done separately, no real need for Counterparty to be involved in this...

**Comment by jdogresorg** [comment]
the original founders wanted a fednode stack which includes a wallet, for the above mentioned reasons. Completely self-contained bitcoin / counterparty / wallet instance that can easily be installed on any computer. I also share this view that the counterparty fednode should contain everything necessary to generate and sign Counterparty transactions, which includes a wallet.

**Comment by jotapea** [comment]
I find the "original founders" answer to be used only when convenient. The original founders also made divisibility immutable, why was that changed then?

Software evolves. Initially it made more sense to have a one-stop solution to use Counterparty. Nowadays, I don't think is the case.

**Comment by jdogresorg** [comment]
> I find the "original founders" answer to be used only when convenient.

sorry you feel that way. When in doubt, I look at what the original founders did, because they have had valid reasons for everything they did. Your more than welcome to reach out to them directly and ask their reasons.

**Comment by jotapea** [comment]
Yes they really did a beautiful work here, and it makes me sad when ugly features make it to the protocol.

Lets see what other people think about maturing the protocol stack by making it leaner...
